runtime/src/main/java/org/corfudb/runtime/clients/BaseHandler.java

```diff
@@ -159,4 +186,182 @@ private static Object handleWrongClusterId(CorfuPayloadMsg<WrongClusterMsg> msg,
+                                                IClientRouter r) {
+        return true;
+    }
+
+    /**
+     * Handle a reset response from the server.
+     *
+     * @param msg The ping response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router
+     * @return Always True, since the reset message was successful.
+     */
+    @ResponseHandler(type = PayloadCase.RESET_RESPONSE)
+    private static Object handleResetResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                              IClientRouter r) {
+        log.info("Received RESET_RESPONSE from the server - {}", msg);
+        return true;
+    }
+
+    /**
+     * Handle a seal response from the server.
+     *
+     * @param msg The ping response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router
+     * @return Always True, since the seal message was successful.
+     */
+    @ResponseHandler(type = PayloadCase.SEAL_RESPONSE)
+    private static Object handleSealResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                             IClientRouter r) {
+        return true;
+    }
+
+    /**
+     * Handle a version response from the server.
+     * For old CorfuMsg, use {@link #handleVersionResponse(JSONPayloadMsg, ChannelHandlerContext, IClientRouter)}
+     *
+     * @param msg The ping response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router
+     * @return The VersionInfo object fetched from response msg.
+     */
+    @ResponseHandler(type = PayloadCase.VERSION_RESPONSE)
+    private static Object handleVersionResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        VersionResponseMsg versionResponseMsg = msg.getPayload().getVersionResponse();
+        return CorfuProtocolBase.getVersionInfo(versionResponseMsg);
+    }
+
+    /**
+     * Handle a UNKNOWN_ERROR response from the server.
+     * For old CorfuMsg, use {@link #handleServerException(CorfuPayloadMsg, ChannelHandlerContext, IClientRouter)}
+     *
+     * @param msg The wrong epoch message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return none, throw a wrong epoch exception instead.
+     */
+    @ServerErrorsHandler(type = ErrorCase.UNKNOWN_ERROR)
+    private static Object handleUnknownError(ResponseMsg msg, ChannelHandlerContext ctx,
+                                             IClientRouter r) throws Throwable {
+        ByteString bs = msg.getPayload().getServerError().getUnknownError().getThrowable();
+        byte[] bytes = new byte[bs.size()];
+        bs.copyTo(bytes, 0);
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes)) {
+            try (ObjectInputStream ois = new ObjectInputStream(bis)) {
+                throw (Throwable) ois.readObject();
```

Comment:
Is it ok to throw.. the method appears to want to return an Object right?

Reply:
Yeah, I think the signature is to align with the FunctionalInterface defined in ClientResponseHandler, but all the handlers for ServerErrors will throw an exception and we always want the router to invoke completeExceptionally() under these cases.